app/services/order_adjustments_fetcher.rb

```diff
@@ -0,0 +1,66 @@
+class OrderAdjustmentsFetcher
+  include AdjustmentScopes
+
+  def initialize(order)
+    @order = order
+  end
+
+  def admin_and_handling_total
+    admin_and_handling_fees.map(&:amount).sum
+  end
+
+  def payment_fee
+    sum_adjustments "payment_fee"
+  end
+
+  def ship_total
+    sum_adjustments "shipping"
+  end
+
+  def line_item_adjustments(line_item)
+    if adjustments.loaded?
+      adjustments.select{ |adjustment| adjustment.source_id == line_item.id }
+    else
+      adjustments.where(source_id: line_item.id)
+    end
+  end
+
+  private
+
+  attr_reader :order
+
+  def adjustments
+    order.adjustments
+  end
+
+  def sum_adjustments(scope)
+    collect_adjustments(scope).map(&:amount).sum
+  end
+
+  def collect_adjustments(scope)
+    if adjustments.loaded?
+      scope_hash = const_get("#{scope.upcase}_SCOPE")
+
+      adjustments.select do |adjustment|
+        adjustment.public_send(scope_hash.keys.first) == scope_hash.values.first
+      end
+    else
+      adjustments.public_send scope
+    end
+  end
+
+  def admin_and_handling_fees
+    if adjustments.loaded?
+      adjustments.select do |adjustment|
+        scope_hash = ELIGIBLE_SCOPE
+
+        adjustment.public_send(scope_hash.keys.first) == scope_hash.values.first &&
```

Comment:
this scope_hash trick is not obvious to understand, maybe this can extracted to a separate method with a good name in some way?
name of the method: what is this == condition doing? 
maybe also a variable with a good name for scope_hash.keys.first

Reply:
I've extracted a couple of methods here to make it more readable.

```ruby
def adjustments_eager_loaded?
    adjustments.loaded?
end

def match_by_scope(adjustment, scope)
    adjustment.public_send(scope.keys.first) == scope.values.first
end
```